bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -66,165 +60,207 @@
+                    this.localizedFragmentVariationPath = StringUtils.join(localizedXfRootPath, xfRelativePath, PATH_DELIMITER_CHAR, NN_CONTENT);
+                }
             }
 
-            if (currentPage != null) {
-                resolveLocalizedFragmentVariationPath();
-                resolveName(pageManager);
-                retrieveExperienceFragmentChildModels();
-            } else {
-                LOGGER.error("Could not resolve currentPage!");
+            String xfContentPath = String.join(Character.toString(PATH_DELIMITER_CHAR), fragmentVariationPath, NN_CONTENT);
+            if (!resourceExists(localizedFragmentVariationPath) && resourceExists(xfContentPath)) {
+                this.localizedFragmentVariationPath = xfContentPath;
+            }
+            if (!isExperienceFragmentVariation(localizedFragmentVariationPath)) {
+                this.localizedFragmentVariationPath = null;
             }
         }
-
-        appendCssClassNames();
-    }
-
-    @Override
-    public String getLocalizedFragmentVariationPath() {
-        return localizedFragmentVariationPath;
+        return this.localizedFragmentVariationPath;
     }
 
     @Override
     @JsonIgnore
+    @Nullable
     public String getName() {
-        return name;
+        if (this.name == null) {
+            this.name = Optional.ofNullable(this.request.getResourceResolver().adaptTo(PageManager.class))
+                .flatMap(pm -> Optional.ofNullable(this.getLocalizedFragmentVariationPath())
+                    .map(pm::getContainingPage))
+                .map(Page::getParent)
+                .map(Page::getName)
+                .orElse(null);
+        }
+        return this.name;
     }
 
     @NotNull
-    @Override public String getExportedType() {
-        return request.getResource().getResourceType();
+    @Override
+    public String getExportedType() {
+        return this.request.getResource().getResourceType();
     }
 
-    @Nonnull
+    @NotNull
     @Override
     public Map<String, ? extends ComponentExporter> getExportedItems() {
-        return children;
+        if (this.children == null) {
+            this.children = Optional.ofNullable(this.getLocalizedFragmentVariationPath())
+                .filter(StringUtils::isNotBlank)
+                .map(this.request.getResourceResolver()::getResource)
+                .map(Resource::listChildren)
+                .map(it -> ContentFragmentUtils.getComponentExporters(it, this.modelFactory, this.request))
+                .orElseGet(LinkedHashMap::new);
+        }
+        return this.children;
     }
 
-    @Nonnull
+    @NotNull
     @Override
     public String[] getExportedItemsOrder() {
-        return children.isEmpty() ?
-                new String[0] : children.keySet().toArray(new String[children.size()]);
+        return this.getExportedItems().keySet().toArray(new String[0]);
```

Comment:
`getExportedItems()` does not guarantee you get a `LinkedHashMap` that gives the desired order. I would rather we create a private/protected `getChildren` that returns a `LinkedHashMap` and takes care of populating/caching children resolution. `getExportedItems` and `getExportedItemsOrder` can then leverage that method.

Reply:
I don't see a need to introduce a `getChildren` as `children` is a `LinkedHashMap ` and the logic to populate it is based on the newly introduced `ContentFragmentUtils. getComponentExporters` which returns a `LinkedHashMap`.